### PR TITLE
Add new C API `pg_GetPkgdataResource`

### LINKED ISSUE #2221: Add `pgPkgdata_GetResource()` to C API
To load default icon for window/display or default font for font/freetype.

### PR DESCRIPTION
**Pull request by yunline**
fixes #2221

- New API `pg_GetPkgdataResource`
- New API `pgImage_LoadBasic`

### CODE REVIEW

**Comment by narilee2006** [comment]
What is pg_GetPkgdataResource ?

**Comment by yunline** [comment]
> Do these changes provide any benefits that I'm missing? I can see that it makes the code a bit cleaner

You are right. It just cleans up some code.

File: src_c/display.c [hunk 1]
```diff
@@ -1323,7 +1248,8 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
 
     /*set the window icon*/
     if (!state->icon) {
-        state->icon = pg_display_resource(icon_defaultname);
+        PyObject *icon_fileobj = pg_GetPkgdataResource(icon_defaultname);
```
Comment (Starbuck5) [inline src_c/display.c:1251]: This isn't a "file object," I believe, so the variable is misleadingly named.

File: src_c/image.c [hunk 2]
```diff
@@ -66,12 +67,12 @@ find_extension(const char *fullname)
 }
 
 static PyObject *
-image_load_basic(PyObject *self, PyObject *obj)
+pgImage_LoadBasic(PyObject *fileobj)
```
Comment (Starbuck5) [inline src_c/image.c:70]: Same comment about `fileobj`

Comment (yunline) [reply src_c/image.c:70]: I don't think this one is misleading.
`fileobj` can be a str filename or a file-like object.

File: src_c/display.c (hunk 1 above)
Comment (yunline) [reply src_c/display.c]: fixed

File: src_c/base.c [hunk 3]
```diff
@@ -2064,6 +2064,79 @@ pg_uninstall_parachute(void)
...
+    else if (encoded != Py_None) {
+        Py_DECREF(fresult);
+        fresult = encoded;
+    }
+    else {
+        Py_DECREF(encoded);
+    }
+
+    // return fresult
+    goto get_resource_end;
```
Comment (ankith26) [inline src_c/base.c:2131]: this is a redundant `goto`, even without it, control is going to flow into the block below

File: src_c/window.c [hunk 4]
```diff
@@ -742,7 +665,8 @@ window_init(pgWindowObject *self, PyObject *args, PyObject *kwargs)
 
     SDL_SetWindowData(_win, "pg_window", self);
 
-    PyObject *icon = pg_display_resource(icon_defaultname);
+    PyObject *icon_path_obj = pg_GetPkgdataResource(icon_defaultname);
+    PyObject *icon = pgImage_LoadBasic(icon_path_obj);
```
Comment (ankith26) [inline src_c/window.c:669]: Needs check for `icon_path_obj` being `NULL` (and exit early with error forward) and if it is not `NULL` it needs to be `DECREF`d after `pgImage_LoadBasic` finishes

File: src_c/display.c [hunk 5]
```diff
@@ -1334,7 +1259,8 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
 
     /*set the window icon*/
     if (!state->icon) {
-        state->icon = pg_display_resource(icon_defaultname);
+        PyObject *icon_path_obj = pg_GetPkgdataResource(icon_defaultname);
+        state->icon = pgImage_LoadBasic(icon_path_obj);
```
Comment (ankith26) [inline src_c/display.c:1263]: same as above review (except here maybe just ignore the error and don't even call `pgImage_LoadBasic` in that case)

**Review by ankith26 — CHANGES_REQUESTED** [review]

**Comment by MyreMylar** [comment]
This could use merging with main as the conflicts are making the diff hard to understand now.

**Review by MyreMylar — CHANGES_REQUESTED** [review]
Please merge with main and resolve the conflicts, to make it easier to review.

**Comment by yunline** [comment]
It‘s not worth to create two C APIs and just simplify code a little bit.
The simplifying works will contunue in #2552

**Pull request closed without merging**